changes/3904.changed

```diff
@@ -0,0 +1,6 @@
+Restyled the pagination section.
+Added proper spacing/alignment to pagination section.
+Modified the back "<" and next ">" arrows to navigate to the previous/next page, respectively, rather than the first/last page.
+The back "<" and next ">" arrows no longer show when they aren't relevant (i.e. when on the first/last page).
+Added Input component for changing the page number, in addition to using the arrows.
+Added total number of rows to footer.
```

Comment:
Please follow the `Added`, `Changed`, `Fixed`, `Removed` convention of files and sentence structure: https://docs.nautobot.com/projects/core/en/stable/release-notes/version-1.5/#v1521-2023-06-12

The file suffix let's Towncrier know which section it should live in.